### DISCUSSION TITLE
TLS alert: 'HandshakeFailure'

### DISCUSSION THREAD

**Discussion by LincolnDing** [betalgo/openai#99] [Q&A]
Hi,this is code piece

public async Task<string> GetAnswer(string text)
        {

            string answer = "";

            OpenAIService service = new OpenAIService(new OpenAiOptions() { ApiKey = OpenAIToken });

            CompletionCreateRequest createRequest = new CompletionCreateRequest()
            {

                Prompt = text,
                Temperature = 0.3f,
                MaxTokens = 1000
            };

            var res = await service.Completions.CreateCompletion(createRequest, Models.TextDavinciV3);

            if (res.Successful)
            {
                answer = res.Choices.FirstOrDefault().Text;

            }
            return answer;

        }

When I am running it on my server, it returns an error : Authentication failed because the remote party sent a TLS alert: 'HandshakeFailure'.

When I am running it on local  pc,it runs ok.
I have checked the setting of TLS between server and local pc, they are the same each other.
btw,I can get   normally response from  openai.com’s playground   on server.

thank you.

**Comment by kayhantolga** [comment]
It seems like an issue with your environment. I will move this issue into discussions and if you can share more details about the error we may help you.

**Reply by LincolnDing** [reply]
environment :Windows server 2012 R2  x64.

**Comment by Yookey** [comment]
hi there:
I seem to have the same problem,
**system :** Windows server 2012 R2 x64; **IIS:** 8.5; **TLS:** 1.2
throw an exception:"The SSL connection could not be established, see inner exception."

**Comment by kayhantolga** [comment]
Do you have any more detail, a stack trace maybe?

I will try to add ignore Certificate Errors option to the library.

**Comment by kayhantolga** [comment]
Could you try this code on your server, if it works I will add an option for ignoring the cert error.

```csharp
var clientHandler = new HttpClientHandler();
clientHandler.ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) => true;
var client = new HttpClient(clientHandler);

var options = Options.Create<OpenAiOptions>(new OpenAiOptions()
{
    ApiKey =  Environment.GetEnvironmentVariable("MY_OPEN_AI_API_KEY")!
});
var openAIService = new OpenAIService(client, options);
```

**Reply by kayhantolga** [reply]
if you were using dependency injection and testing would be hard, just let me know. we can find another way to try

**Reply by Yookey** [reply]
Hi kayhantolga:
thank you for your reply.
unfortunately, this code doesn't work, so I upgraded my operating system. and now it works perfectly.
thank you!
